modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -51,41 +52,28 @@ CoreMeshGenerator::validParams()
 }
 
 CoreMeshGenerator::CoreMeshGenerator(const InputParameters & parameters)
-  : MeshGenerator(parameters),
+  : ReactorGeometryMeshBuilderBase(parameters),
     _inputs(getParam<std::vector<MeshGeneratorName>>("inputs")),
     _empty_key(getParam<MeshGeneratorName>("dummy_assembly_name")),
     _pattern(getParam<std::vector<std::vector<unsigned int>>>("pattern")),
     _extrude(getParam<bool>("extrude"))
 {
-  /* Below, we get the ReactorMeshParams info from one of the pin inputs.
-     The reactor params gives the axial info if we are to extrude as well as the assembly pitch
-     that the pin pattern is to be checked against.
-     We then create the PatternedMesh subgenerators and pass the parameter info needed for the
-     inputs. Finally, if the mesh is to be extruded we create the FancyExtruderGenerator
-     subgenerator and the PlaneIDMeshGenerator subgenerator to handle this.
-  */
-  MeshGeneratorName _reactor_params =
+  // Initialize name_id_map and current_block_id from ReactorMeshParams object stored
+  // in pin input
+  MeshGeneratorName reactor_params =
       MeshGeneratorName(getMeshProperty<std::string>("reactor_params_name", _inputs[0]));
```

Comment:
seems odd / not intuitive that only the first one is queried, other reactor info stored in the other ones is ignored

Reply:
I could add a check to make sure all values of "reactor_params_name" in `_inputs` are the same? They should all be equivalent

Reply:
umm actually probably not. 

This reactor_params mesh generator name, it s the name of that MG at the top of the file right? 
So we could see a user define two if they're being hacky. 

let's just add a check then

Reply:
Check has been added